apps/air-discount-scheme/backend/src/app/modules/discount/discount.service.ts

```diff
@@ -40,22 +50,29 @@ export class DiscountService {
     const discountCode = this.generateDiscountCode()
     const discountCodeCacheKey = this.getDiscountCodeCacheKey(discountCode)
     const nationalIdCacheKey = this.getNationalIdCacheKey(nationalId)
-    await this.cacheManager.set(
-      discountCodeCacheKey,
-      { nationalId },
-      { ttl: ONE_DAY },
-    )
-    await this.cacheManager.set(
-      nationalIdCacheKey,
-      { discountCode },
-      { ttl: ONE_DAY },
-    )
+    const id = uuid()
+    const ttl = { ttl: ONE_DAY }
+    await this.cacheManager.set(id, { nationalId, discountCode }, ttl)
+    await this.cacheManager.set(discountCodeCacheKey, id, ttl)
+    await this.cacheManager.set(nationalIdCacheKey, id, ttl)
     return new Discount(discountCode, nationalId, ONE_DAY)
   }
 
+  private async getValue(cacheKey: string): Promise<CachedDiscount> {
+    const cacheId = await this.cacheManager.get(cacheKey)
+    if (!cacheId) {
+      return null
+    }
+
+    const cacheValue = await this.cacheManager.get(cacheId)
+    if (!cacheValue) {
+      return null
+    }
```

Comment:
Missing a return here (the cacheValue is never returned). Can most likely skip the if statement:
```typescript
  private async getValue(cacheKey: string): Promise<CachedDiscount> {
    const cacheId = await this.cacheManager.get(cacheKey)
    if (!cacheId) {
      return null
    }

    return this.cacheManager.get(cacheId)
```